examples/plot_int_alignment.py

```diff
@@ -0,0 +1,216 @@
+#   - target test: sub-07 PA contrasts, used as a ground truth to score predictions
+# We make a single 4D Niimg from our list of 3D filenames
+target_train = df[df.subject == "sub-07"][df.acquisition == "ap"].path.values
+target_train = concat_imgs(target_train)
+target_train_data = masker.transform(target_train)
+target_test = df[df.subject == "sub-07"][df.acquisition == "pa"].path.values
+
+
+###############################################################################
+# Compute a baseline (average of subjects)
+# ----------------------------------------
+# We create an image with as many contrasts as any subject representing for
+# each contrast the average of all train subjects maps.
+#
+
+import numpy as np
+
+masked_imgs = [masker.transform(img) for img in template_train]
+average_img = np.mean(masked_imgs[:-1], axis=0)
+average_subject = masker.inverse_transform(average_img)
+
+###############################################################################
+# Create a template from the training subjects.
+# ---------------------------------------------
+# We define an estimator using the class TemplateAlignment:
+#   * We align the whole brain through multiple local alignments.
+#   * These alignments are calculated on a parcellation of the brain in 100 pieces,
+#     this parcellation creates group of functionnally similar voxels.
+#   * The template is created iteratively, aligning all subjects data into a
+#     common space, from which the template is inferred and aligning again to this
+#     new template space.
+#
+
+from nilearn.image import index_img
+from fmralign.alignment_methods import IndividualizedNeuralTuning
+from fmralign.hyperalignment.piecewise_alignment import PiecewiseAlignment
+from fmralign.hyperalignment.regions import compute_parcels
+
+###############################################################################
+# Predict new data for left-out subject
+# -------------------------------------
+# We use target_train data to fit the transform, indicating it corresponds to
+# the contrasts indexed by train_index and predict from this learnt alignment
+# contrasts corresponding to template test_index numbers.
+# For each train subject and for the template, the AP contrasts are sorted from
+# 0, to 53, and then the PA contrasts from 53 to 106.
+#
+
+train_index = range(53)
+test_index = range(53, 106)
+
+denoising_data = np.array(masked_imgs)[:, train_index, :]
+training_data = np.array(masked_imgs)[:-1]
+target_test_masked = np.array(masked_imgs)[:, test_index, :]
+
+
+parcels = compute_parcels(niimg=template_train[0], mask=masker, n_parcels=100, n_jobs=5)
+denoiser = PiecewiseAlignment(n_jobs=5)
+denoised_signal = denoiser.fit_transform(X=denoising_data, regions=parcels)
+target_denoised_data = denoised_signal[-1]
+model = IndividualizedNeuralTuning(
+    parcels=parcels,
+)
+model.fit(training_data, verbose=False)
+stimulus_ = np.copy(model.shared_response)
```

Comment:
Is there a reason to create this copy ? It doesn't look like we're modifying `model.shared_response` (or `stimulus_`). I'm not sure if `stimulus_` is more commonly used in the INT literature ?

Reply:
In fact *stimulus* is the way to call the shared response in the INT literature !